lib/pengine/unpack.c

```diff
@@ -496,21 +496,20 @@ pe_create_node(const char *id, const char *uname, const char *type,
         new_node->priv->variant = pcmk__node_variant_remote;
         pcmk__set_scheduler_flags(scheduler, pcmk__sched_have_remote_nodes);
 
-    } else {
-        /* @COMPAT 'ping' is the default for backward compatibility, but it
-         * should be changed to 'member' at a compatibility break
-         */
-        if (!pcmk__str_eq(type, PCMK__VALUE_PING, pcmk__str_casei)) {
-            pcmk__config_warn("Node %s has unrecognized type '%s', "
-                              "assuming '" PCMK__VALUE_PING "'",
-                              pcmk__s(uname, "without name"), type);
-        }
+    } else if (pcmk__str_eq(type, PCMK__VALUE_PING, pcmk__str_casei)) {
         pcmk__warn_once(pcmk__wo_ping_node,
                         "Support for nodes of type '" PCMK__VALUE_PING "' "
                         "(such as %s) is deprecated and will be removed in a "
                         "future release",
                         pcmk__s(uname, "unnamed node"));
         new_node->priv->variant = pcmk__node_variant_ping;
+
+    } else {
+        // @COMPAT Throw error, with XSL transform for old configs?
```

Comment:
yep. we can deprecate it now and leave it for next time

to prepare for that, we should move the type checking before allocating the new node

a parse_node_variant function would be nice but not necessary now

Reply:
My main question here is whether to convert invalid node type to member going forward (instead of ping previously), or throw an error and skip creating a node at all. It sounds like the latter.

In that case, our XSL transform would not only convert `ping` to `member` but also convert anything other than `member` and `remote` to `member`.

Reply:
I'm thinking we should treat invalid types as member (for 3.0.0), with a deprecation warning that such nodes will be ignored in a future release. I wouldn't convert them via XSL yet, that can happen when we switch that behavior. (It gives people a chance to notice and correct manual edit typos, not that those are likely.)

Reply:
Before 3.0.0, nodes with invalid `type` would have been treated as ping nodes, which can't run resources (not sure if there was any other implication of ping nodes)... which suggests that nodes with invalid `type` would have been useless in practice for a long time. Am I mistaken?

There hasn't been a deprecation warning that they'll be ignored yet, but there has been an "unrecognized type" warning logged at least.

Reply:
Yeah we might as well ignore invalid nodes now. No point in dragging that one out